doc/release/upcoming_changes/15217.deprecation.rst

```diff
@@ -0,0 +1,11 @@
+Passing ``shape=0`` to factory functions in ``numpy.rec`` is deprecated
+-----------------------------------------------------------------------
+
+``0`` is treated as a special case by these functions, which aliases to
+``None``. In future, ``0`` will not be a special case, and will be treated
+as an array length like any other integer is. The affected functions are:
+
+* `numpy.core.records.fromarrays`
+* `numpy.core.records.fromrecords`
+* `numpy.core.records.fromstring`
+* `numpy.core.records.fromfile`
```

Comment:
I think this broke towncrier generation of the release notes since the template does not have a blank line between the end of this bullet list and the link to the PR, it is rendered as
```
as an array length like any other integer is. The affected functions are:

* `numpy.core.records.fromarrays`
* `numpy.core.records.fromrecords`
* `numpy.core.records.fromstring`
* `numpy.core.records.fromfile`
(`gh-15217 <https://github.com/numpy/numpy/pull/15217>`__)

```

Not sure how this passed CI ...

Reply:
Possible that the tests ran when the extra blank line was around (i.e. we merge the ragged-array PR again)